### PR TITLE
Check if image exists before validating

### LINKED ISSUE #5230: Uploading a product picture without a file rasies an exception
### Steps to reproduce the problem
1. Upload an image without sending a file through `productImageCreate`
```
Traceback (most recent call last):
  File "./saleor/graphql/core/mutations.py", line 279, in mutate
    response = cls.perform_mutation(root, info, **data)
  File "./saleor/graphql/product/mutations/products.py", line 1400, in perform_mutation
    validate_image_file(image_data, "image")
  File "./saleor/graphql/core/utils/__init__.py", line 33, in validate_image_file
    if not file.content_type.startswith("image/"):
AttributeError: 'NoneType' object has no attribute 'content_type'
```

### What I expected to happen
A mutation error.

### PR DESCRIPTION
**Pull request by kswiatek92**
closes #5230 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/core/utils/__init__.py [hunk 1]
```diff
@@ -36,6 +36,11 @@ def str_to_enum(name):
 
 def validate_image_file(file, field_name):
     """Validate if the file is an image."""
+    print(type(file))
```
Comment (maarcingebala) [inline saleor/graphql/core/utils/__init__.py:39]: ☠️

File: saleor/graphql/core/utils/__init__.py [hunk 2]
```diff
@@ -36,6 +36,11 @@ def str_to_enum(name):
...
+    if not file:
+        raise ValidationError(
+            {field_name: ValidationError("File is required", code="invalid")}
```
Comment (maarcingebala) [inline saleor/graphql/core/utils/__init__.py:42]: Not sure what's the best way to use error codes as enums here so we can keep it for now, but the error code should be `required`.

File: saleor/graphql/core/utils/__init__.py (hunk 1 above)
Comment (NyanKiyoshi) [reply saleor/graphql/core/utils/__init__.py:39]: Smells like a type hint would be welcome

File: CHANGELOG.md [hunk 3]
```diff
@@ -88,6 +88,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Add shipping rate price validation - #5411 by @kswiatek92
 - Remove unused settings and environment variables - #5420 by @maarcingebala
 - Add product price validation - #5413 by @kswiatek92
+- Check if image exists befeore validating - #5425 by @kswiatek92
```
Comment (NyanKiyoshi) [inline CHANGELOG.md:91]:
```suggestion
- Check if image exists before validating - #5425 by @kswiatek92
```

File: saleor/graphql/core/utils/__init__.py [hunk 4]
```diff
@@ -36,6 +36,10 @@ def str_to_enum(name):
 
 def validate_image_file(file, field_name):
     """Validate if the file is an image."""
+    if not file:
```
Comment (NyanKiyoshi) [inline saleor/graphql/core/utils/__init__.py:39]: Type hint?

Comment (kswiatek92) [reply saleor/graphql/core/utils/__init__.py:39]: I tried `UploadedFile`, and `InMemoryUploadedFile`
but mypy rasied `Item "None" of "Optional[str]" has no attribute "startswith"`

File: CHANGELOG.md [hunk 5]
```diff
@@ -89,6 +89,8 @@ All notable, unreleased changes to this project will be documented in this file.
 - Remove unused settings and environment variables - #5420 by @maarcingebala
 - Add product price validation - #5413 by @kswiatek92
 - Add attribute validation to attributeAssign - #5423 by @kswiatek92
+- Check if image exists befeore validating - #5425 by @kswiatek92
+- Check if image exists before validating - #5425 by @kswiatek92
```
Comment (NyanKiyoshi) [inline CHANGELOG.md:93]:
```suggestion
- Check if image exists before validating - #5425 by @kswiatek92
```

File: saleor/graphql/core/utils/__init__.py (hunk 4 above)
Comment (NyanKiyoshi) [reply saleor/graphql/core/utils/__init__.py:39]: Probably because you need to put it as `Optional`?

Comment (NyanKiyoshi) [reply saleor/graphql/core/utils/__init__.py:39]: Or maybe because `content_type` is nullable (implicitly/explicitly)?

Comment (kswiatek92) [reply saleor/graphql/core/utils/__init__.py:39]: `content_type` can be None

**Pull request merged**